component/s3storage/s3storage_test.go

```diff
@@ -2263,44 +2294,37 @@ func (s *s3StorageTestSuite) TestReadLinkError() {
 	s.assert.EqualValues(syscall.ENOENT, err)
 }
 
+func (s *s3StorageTestSuite) TestReadLinkDisabled() {
+	defer s.cleanupTest()
+	// Setup
+	name := generateFileName()
+
+	_, err := s.s3Storage.ReadLink(internal.ReadLinkOptions{Name: name})
+	s.assert.Error(err)
+	s.assert.EqualValues(syscall.ENOTSUP, err)
+}
+
 func (s *s3StorageTestSuite) TestGetAttrDir() {
 	defer s.cleanupTest()
-	vdConfig := generateConfigYaml(storageTestConfigurationParameters)
```

Comment:
When I was figuring out how to change the config to enable symlinks in the tests that need them, I noticed that `setupTestHelper` sets some config parameters by default when an empty config string is passed in. To do that, it uses `generateConfigYaml(storageTestConfigurationParameters)`... but that is exactly the same as `vdConfig`.
I realized the reason for this is that we copied these tests from `azstorage`, where `vdConfig` was used to set `virtual-directory` to `true`.
But `s3storage` doesn't have a `virtual-directory` setting, so that's why these `vdConfig` variables are just redundantly testing the same default settings a second time.
Once I figured all that out, I decided to tear out all the virtual directory stuff from the tests in `s3storage`.